diff --git a/node_modules/image-size/dist/index.cjs b/node_modules/image-size/dist/index.cjs
index bde0210..3681c92 100644
--- a/node_modules/image-size/dist/index.cjs
+++ b/node_modules/image-size/dist/index.cjs
@@ -163,6 +163,7 @@ var HEIF = {
     while (currentOffset < ipcoBox.offset + ipcoBox.size) {
       const ispeBox = findBox(input, "ispe", currentOffset);
       if (!ispeBox) break;
+      if (ispeBox.size <= 0) break;
       const rawWidth = readUInt32BE(input, ispeBox.offset + 12);
       const rawHeight = readUInt32BE(input, ispeBox.offset + 16);
       const clapBox = findBox(input, "clap", currentOffset);
@@ -253,6 +254,7 @@ var ICNS = {
     const images = [];
     while (imageOffset < fileLength && imageOffset < inputLength) {
       const imageHeader = readImageHeader(input, imageOffset);
+      if (imageHeader[1] <= 0) break;
       const imageSize2 = getImageSize2(imageHeader[0]);
       images.push(imageSize2);
       imageOffset += imageHeader[1];
@@ -487,6 +489,7 @@ function extractPartialStreams(input) {
   while (offset < input.length) {
     const jxlpBox = findBox(input, "jxlp", offset);
     if (!jxlpBox) break;
+    if (jxlpBox.size <= 0) break;
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
